parser-typechecker/src/Unison/TermParser.hs

```diff
@@ -183,7 +184,7 @@ parsePattern =
         f patterns vs =
           let loc = foldl (<>) (ann tok) $ map ann patterns
           in (Pattern.Constructor loc ref cid patterns, vs)
-    unzipPatterns f <$> many (leaf <|> seqLiteral)
+    unzipPatterns f <$> many patternCandidates
```

Comment:
@aryairani FYI I believe this change introduced #1640 - by having the arguments to the constructor also include the `constructor` parser, you end up with it being right associated, where `Foo Bar a b` is parsed as `Foo (Bar a b)`. The fix I implemented to avoid nullary constructors being treated as variables was to just add a nullary constructor parser and move that to before the variable parser in `leaf`. See this note here: 

https://github.com/unisonweb/unison/pull/1642/files#diff-211d62c2ee7dccd868dc459a4e36136cR159-R164

Sorry for not spotting this during review of this PR!

It was good that we had the #868 test case though, because I did it incorrectly at first and got a failure that led to me figuring this out. :)